packages/caip/src/caip2/caip2.ts

```diff
@@ -32,72 +32,84 @@ type ToCAIP2Args = {
-      namespace: ChainNamespace.Cosmos,
-      reference: {
-        [NetworkTypes.COSMOSHUB_MAINNET]: ChainReference.CosmosHubMainnet,
-        [NetworkTypes.COSMOSHUB_VEGA]: ChainReference.CosmosHubVega,
-
-        [NetworkTypes.OSMOSIS_MAINNET]: ChainReference.OsmosisMainnet,
-        [NetworkTypes.OSMOSIS_TESTNET]: ChainReference.OsmosisTestnet
-      }
+const shapeShiftToCAIP2 = Object.freeze({
+  [ChainTypes.Ethereum]: {
+    namespace: ChainNamespace.Ethereum,
+    reference: {
+      [NetworkTypes.MAINNET]: ChainReference.EthereumMainnet,
+      [NetworkTypes.ETH_ROPSTEN]: ChainReference.EthereumRopsten,
+      [NetworkTypes.ETH_RINKEBY]: ChainReference.EthereumRinkeby
+    }
+  },
+  [ChainTypes.Bitcoin]: {
+    namespace: ChainNamespace.Bitcoin,
+    reference: {
+      [NetworkTypes.MAINNET]: ChainReference.BitcoinMainnet,
+      [NetworkTypes.TESTNET]: ChainReference.BitcoinTestnet
+    }
+  },
+  [ChainTypes.Cosmos]: {
+    namespace: ChainNamespace.Cosmos,
+    reference: {
+      [NetworkTypes.COSMOSHUB_MAINNET]: ChainReference.CosmosHubMainnet,
+      [NetworkTypes.COSMOSHUB_VEGA]: ChainReference.CosmosHubVega
     }
-  } as const
+  },
+  [ChainTypes.Osmosis]: {
+    namespace: ChainNamespace.Cosmos,
+    reference: {
+      [NetworkTypes.OSMOSIS_MAINNET]: ChainReference.OsmosisMainnet,
+      [NetworkTypes.OSMOSIS_TESTNET]: ChainReference.OsmosisTestnet
+    }
+  }
+})
 
+export const toCAIP2: ToCAIP2 = ({ chain, network }): string => {
   const namespace: ChainNamespace = shapeShiftToCAIP2[chain].namespace
 
   switch (chain) {
-    case ChainTypes.Cosmos: {
+    case ChainTypes.Ethereum: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
-        case NetworkTypes.COSMOSHUB_MAINNET:
-        case NetworkTypes.COSMOSHUB_VEGA:
-        case NetworkTypes.OSMOSIS_MAINNET:
-        case NetworkTypes.OSMOSIS_TESTNET: {
+        case NetworkTypes.MAINNET:
+        case NetworkTypes.ETH_ROPSTEN:
+        case NetworkTypes.ETH_RINKEBY: {
           const reference: ChainReference = referenceMap[network]
           const caip2 = `${namespace}:${reference}`
           return caip2
         }
       }
       break
     }
-
-    case ChainTypes.Ethereum: {
+    case ChainTypes.Bitcoin: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
         case NetworkTypes.MAINNET:
-        case NetworkTypes.ETH_ROPSTEN:
-        case NetworkTypes.ETH_RINKEBY: {
+        case NetworkTypes.TESTNET: {
           const reference: ChainReference = referenceMap[network]
           const caip2 = `${namespace}:${reference}`
           return caip2
         }
       }
       break
     }
-
-    case ChainTypes.Bitcoin: {
+    case ChainTypes.Cosmos: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
-        case NetworkTypes.MAINNET:
-        case NetworkTypes.TESTNET: {
+        case NetworkTypes.COSMOSHUB_MAINNET:
```

Comment:
Has there been discussion on how mainnet will be managed with cosmos upgrades? Will `COSMOSHUB_MAINNET` just keep changing the underlying value once version is upgraded from v4? Would we want to be explicit and just name it `COSMOSHUB_MAINNET_4`? Will there be any need to reference a previous mainnet version where this would be useful (previous tx history)?

Reply:
No, but I didn't add this enum. It's already in `main`. If there should be a change to this I feel that's outside the scope of the intended changes for this PR specifically.

Reply:
Well maybe quick poll on follow up discussion?

Reply:
I don't understand the question.

Reply:
> Will there be any need to reference a previous mainnet version where this would be useful (previous tx history)?

I'm not sure that's something we would need. If you try to access previous mainnets on Figment's [Hubble](https://hubble.figment.io/cosmos/chains/cosmoshub-3), you will have an info message stating that "Chain appears to be stopped. No new blocks for over 1 year!".
My understanding is we should be able to always use the newest chain, but happy to continue this in Discussions.